crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -0,0 +1,121 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Repository
+    attr_reader :platform, :id, :registry
+
+    class << self
+      #
+      # class methods
+      #
+      def registry
+        @@config ||= YAML.load_file(File.join(Rails.root, "config", "repos.yml"))
```

Comment:
BTW I have noticed some not so nice behaviour around the memoization. When I change the e.g. `required` option in the yaml file, the next browser request picks it up but if I continue refreshing it jumps between states again. I think there is some underlying issue. It seems that there is more than one yaml file in memory, and the app refers to both.

I am not sure how long the yaml stays in memory, besides restarting the server. maybe it even has to do with my setup, as i am running in rails development mode.
Unfortunately I don't have a quick way to test it now with the product.

Shall we give up some performance here by not memoizing? or does it make sense to investigate/debug the rails cache?